nltk/parse/malt.py

```diff
@@ -5,245 +6,492 @@
+	(r'(across|Across|through|Through)$', 'IN'),	# space prepopsitions
+	(r'(into|Into|towards|Towards)$', 'IN'),		# space prepopsitions
+	(r'(onto|Onto|from|From)$', 'IN'),				# space prepopsitions
+	(r'.*able$', 'JJ'),                # adjectives
+	(r'.*ness$', 'NN'),                # nouns formed from adjectives
+	(r'.*ly$', 'RB'),                  # adverbs
+	(r'.*s$', 'NNS'),                  # plural nouns
+	(r'.*ing$', 'VBG'),                # gerunds
+	(r'.*ed$', 'VBD'),                 # past tense verbs
+	(r'.*', 'NN'),                     # nouns (default)
+	])
+	return _tagger.tag
+
+
+def find_maltparser(parser_dirname):
+	"""
+	A module to find MaltParser .jar file and its dependencies.
+	"""
+	if os.path.exists(parser_dirname): # If a full path is given.
+		_malt_dir = parser_dirname
+	else: # Try to find path to maltparser directory in environment variables.
+		_malt_dir = find_dir(parser_dirname, env_vars=('MALT_PARSER',))
+	# Checks that that the found directory contains all the necessary .jar 
+	malt_dependencies = ['','','']
+	_malt_jars = set(find_jars_within_path(_malt_dir))
+	_jars = set(jar.rpartition('/')[2] for jar in _malt_jars)
+	malt_dependencies = set([u'log4j.jar', u'libsvm.jar', u'liblinear-1.8.jar'])
+					
+	assert malt_dependencies.issubset(_jars)
+	assert any(filter(lambda i: i.startswith('maltparser-') and \
+								i.endswith('.jar'), _jars))
+	return list(_malt_jars)
+
+def find_malt_model(model_filename):
+	"""
+	A module to find pre-trained MaltParser model.
+	"""
+	if model_filename == None:
+		return 'malt_temp.mco'
+	elif os.path.exists(model_filename): # If a full path is given.
+		return model_filename
+	else: # Try to find path to malt model in environment variables.
+		return find_file(model_filename, env_vars=('MALT_MODEL',), 
+						verbose=False)
+		
+class MaltParser(ParserI):
+	"""
+	A class for dependency parsing with MaltParser. The input is the paths to:
+     - a maltparser directory
+     - (optionally) the path to a pre-trained MaltParser .mco model file
+     - (optionally) the tagger to use for POS tagging before parsing
+     - (optionally) additional Java arguments
+	
+	Example:
+        >>> from nltk.parse import malt
+        >>> # With MALT_PARSER and MALT_MODEL environment set.
+        >>> mp = malt.MaltParser('maltparser-1.7.2', 'engmalt.linear-1.7.mco') # doctest: +SKIP
+        >>> mp.parse_one('I shot an elephant in my pajamas .'.split()).tree() # doctest: +SKIP
+        (shot I (elephant an) (in (pajamas my)) .)
+        >>> # Without MALT_PARSER and MALT_MODEL environment.
+        >>> mp = malt.MaltParser('/home/alvas/maltparser-1.7.2/', '/home/alvas/engmalt.linear-1.7.mco') # doctest: +SKIP
+        >>> mp.parse_one('I shot an elephant in my pajamas .'.split()).tree() # doctest: +SKIP
+        (shot I (elephant an) (in (pajamas my)) .)
+	"""
+	def __init__(self, parser_dirname, model_filename=None, tagger=None, 
+				 additional_java_args=[]):
```

Comment:
Please make ``additional_java_args=None`` and add this
```python
if additional_java_args is None:
    additional_java_args = []
```
as having mutable default parameters might lead to obscure bugs.